middleware/jmx/detectors-jmx.tf

```diff
@@ -0,0 +1,239 @@
+# jmx thread count
+resource "signalfx_detector" "jmx_thread_count" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] JMX Threads Count"
+
+  program_text = <<-EOF
+    signal = data('gauge.jvm.threads.count', filter=${module.filter-tags.filter_custom})${var.jmx_thread_count_aggregation_function}${var.jmx_thread_count_transformation_function}.publish('signal')
+    detect(when(signal > ${var.jmx_thread_count_threshold_critical})).publish('critical')
+    detect(when(signal > ${var.jmx_thread_count_threshold_warning}) and (signal < ${var.jmx_thread_count_threshold_critical})).publish('WARN')
+EOF
+
+  rule {
+    description           = "is too low > ${var.jmx_thread_count_threshold_warning}"
+    severity              = "Warning"
+    detect_label          = "WARN"
+    disabled              = coalesce(var.jmx_thread_count_disabled_warning, var.detectors_disabled)
+    notifications         = coalescelist(var.jmx_thread_count_notifications_warning, var.notifications)
+    parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{readableRule}}} on {{{dimensions}}}"
+  }
+  rule {
+    description           = "is too low > ${var.jmx_thread_count_threshold_critical}"
+    severity              = "Critical"
```

Comment:
I think critical is a too high severity for this detector which is not realy generic.

In my opinion we should lower the severity and keep in enabled by default or disabled it by default but decrease severity